Makefile

```diff
@@ -40,6 +40,10 @@ clean:
 # Building #
 ############
 
+.PHONY: run-local
+run-local: build
+	KUBECONFIG=$(KUBECONFIG) ./hack/local-cmo.sh
+
 .PHONY: build
 build: operator
```

Comment:
Also I had to do the following changes in order to make this work on macOS,

```
diff --git a/hack/local-cmo.sh b/hack/local-cmo.sh
index 51f79f2b..98fdd183 100755
--- a/hack/local-cmo.sh
+++ b/hack/local-cmo.sh
@@ -96,10 +96,9 @@ main(){
   gojsontoyaml -yamltojson  < $operator_config |
       jq -r '.data["metrics.yaml"]' > $telemetry_conf

-  local -a images
-  readarray -t images < <(images_from_deployment)
+  local -a images=$(images_from_deployment)

-  run ./operator "${images[@]}" \
+  run ./operator ${images} \
     -assets assets/ \
     -telemetry-config $telemetry_conf \
     -kubeconfig "${KUBECONFIG}" \
```

Reply:
@arajkumar thanks for trying this out on OSX ... let me see if I can make this work for both. 

I am surprised that osx doesn't have a the shell builtin `readarray` 🤯

we still want to user array for images .. so `"{images[@]}"` should be used. 
I will try and replace readarray with a `while read...`

Reply:
@arajkumar 
> GO_BUILD_RECIPE is set to build the target binary for linux. It would be better to have a separate recipe for local cmo to make the macos users happy 

I agree with you that it is nice to have  but perhaps outside the scope of this patch. 

I see that 84e101dab4f92ca886567f6303d476b74f930edb reverts this functionality that was already in `Makefile`

```diff
-GOOS?=$(shell go env GOOS)
-GOARCH?=$(shell go env GOARCH)
 GO111MODULE?=on
 GOPROXY?=http://proxy.golang.org
 export GO111MODULE
@@ -30,7 +28,7 @@ TOOLING=$(EMBEDMD_BIN) $(GOBINDATA_BIN) $(JB_BIN) $(GOJSONTOYAML) $(JSONNET_BIN)
 JSONNET_SRC=$(shell find ./jsonnet -type f -not -path "./jsonnet/vendor*")
 JSONNET_VENDOR=jsonnet/vendor
·
-GO_BUILD_RECIPE=GOOS=$(GOOS) GOARCH=$(GOARCH) CGO_ENABLED=0 go build --ldflags="-s -X $(GO_PKG)/pkg/operator.Version=$(VERSION)"
+GO_BUILD_RECIPE=GOOS=linux CGO_ENABLED=0 go build --ldflags="-s -X $(GO_PKG)/pkg/operator.Version=$(VERSION)"
·
 
```

Reply:
@dgrisonnet Do you know why the above was reverted in https://github.com/openshift/cluster-monitoring-operator/pull/960/files ?

Reply:
We have a ticket for this that @bison is working on - https://issues.redhat.com/browse/MON-1046

Reply:
@paulfantom , I was talking about only 
```diff
-GO_BUILD_RECIPE=GOOS=$(GOOS) GOARCH=$(GOARCH) CGO_ENABLED=0 go build --ldflags="-s -X $(GO_PKG)/pkg/operator.Version=$(VERSION)"
+GO_BUILD_RECIPE=GOOS=linux CGO_ENABLED=0 go build --ldflags="-s -X $(GO_PKG)/pkg/operator.Version=$(VERSION)"
```
which seems unrelated to https://issues.redhat.com/browse/MON-1046 🤔

Reply:
That's actually related to https://issues.redhat.com/browse/MON-1046, since I brought this improvement to the Makefile build with the fix of this ticket. The original idea behind this ticket was to move the script to golang which is was I was tasked to do. But, the script was buggy, and since we were blocked by it just before a final freeze, so we decided to revert it instead of fixing the bugs. More info here: https://issues.redhat.com/browse/MON-1046?focusedCommentId=15592931&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-15592931. That resulted in the build recipe improvements to also be reverted.

That said, I am totally in favor of adding these env variables again.

Reply:
@paulfantom @dgrisonnet Thank you for clarifying. I think it is a nice improvement and I have already added that to Makefile :). Thanks for doing the ground work. 

@arajkumar thanks for raising the request :)